File: cwclock-mobile/lib/common/project_label.dart
import '../models/client.dart';
import '../models/project.dart';

/// Renders a project as "{client} - {project}" for pickers.
String projectLabel(Project project, List<Client> clients) {
  Client? client;
  for (final c in clients) {
    if (c.id == project.clientId) {
      client = c;
      break;
    }
  }
  return '${client?.name ?? '?'} - ${project.name}';
}

File: cwclock-mobile/src/common/projectLabel.js
// Ported from cwclock-ui/src/Components/common/projectLabel.js - renders a
// project as "{client} - {project}" for pickers.
const projectLabel = (project, clients) => {
  const client = clients.find((c) => c.id === project.clientId);
  return `${client ? client.name : "?"} - ${project.name}`;
};

export default projectLabel;
